src/Makefile

```diff
@@ -340,6 +344,24 @@ ifeq ($(KERNEL),Darwin)
 	LDFLAGS += -arch $(arch) -mmacosx-version-min=10.14
 endif
 
+# To cross-compile for Android, NDK version r21 or later is recommended. 
+# In earlier NDK versions, you'll need to pass -fno-addrsig if using GNU binutils.
+# Currently we don't know how to make PGO builds with the NDK yet.
+ifeq ($(COMP),ndk)
+	CXXFLAGS += -stdlib=libc++ -fPIE -DUSE_POPCNT -DUSE_NEON
+	LDFLAGS += -static-libstdc++ -fPIE -pie -lm -latomic -DUSE_POPCNT -DUSE_NEON -DNDEBUG -O3 -flto
+	LDFLAGS := $(filter-out -lpthread,$(LDFLAGS))
+	ifeq ($(ARCH),armv7)
+	comp=armv7a-linux-androideabi16-clang
+	CXX=armv7a-linux-androideabi16-clang++
+	CXXFLAGS += -mthumb -march=armv7-a -mfloat-abi=softfp 
+	endif
+	ifeq ($(ARCH),armv8)
+	comp=aarch64-linux-android21-clang
```

Comment:
is there a way to avoid hard-coding `android21` so that it will work with another version of ndk ? If not easy, this is a good start anyway.

Reply:
Probably, but I'm not sure how. Sanity check was failing me until I hardcoded it in there. 

API level 21 corresponds to Android 5.0 Lollipop. It's the earliest version with a 64-bit support. Targeting it should allow for the widest support. Targeting a later API may or may not improve performance, but I can see someone wishing to target their own device as well.

Reply:
Ok, I don't know the details of that. This might be a good choice. However, in the future, with later ndks maybe the android21 binary won't be part of the ndk, and the code would fail to compile.

Reply:
I see. I don't think we need to be very concerned about future compatibility. While Android/NDK will drop support for older APIs over time, I doubt they'll abandon the current structure of the NDK any time soon. They'll stick with the LLVM/clang ecosystem.

All we could need might be as simple as changing/updating the version numbers of the triple. It should stay `aarch64-linux-android28-` and so on.

Reply:
Also, the `android21` there stands for the target API Level, and isn't hardwired to the NDK r21. They both happen to be 21 by coincidence. 

https://source.android.com/setup/start/build-numbers

The oldest target still supported in NDK r21d is API Level 16 (Jelly Bean 4.1.x from 2012-2013).

Reply:
so in this sense that's not to different from '-mmacosx-version-min=10.14' which we already have, and one day will probably also fail to compile.